src/tools/components/MSAView.tsx

```diff
@@ -123,17 +129,11 @@ const MSAView: FC<MSAViewProps> = ({
     [features, annotation]
   );
 
-  const setOverviewTrackData = useCallback(
-    (node): void => {
-      if (node) {
-        // eslint-disable-next-line no-param-reassign
-        node.data = getFullAlignmentSegments(alignment)
-          .map(({ trackData }) => trackData)
-          .flat();
-      }
-    },
-    [alignment]
-  );
+  useEffect(() => {
+    if (alignment) {
+      setOverviewTracksData(getFullAlignmentSegments(alignment));
```

Comment:
This is not async nor resulting from a user action, so that should'nt really be in a useEffect and setting as state directly dependent of a prop.
That should be in the main part of the render function, and if the calculation is expensive, that should be wrapped by a useMemo